### PR TITLE
Mock locator

### PR DESCRIPTION
**Pull request by wli3**

### CODE REVIEW

File: src/Microsoft.DotNet.TemplateLocator/ITemplateLocator.cs [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.TemplateLocator
+{
+    public interface ITemplateLocator
+    {
+        public IReadOnlyCollection<IOptionalSDKTemplateInfo> GetDotnetSdkTemplatesFromSDKversion(string SDKversion);
+    }
```
Comment (wli3) [inline src/Microsoft.DotNet.TemplateLocator/ITemplateLocator.cs:11]: Add get SDK version method

File: src/Microsoft.DotNet.TemplateLocator/IOptionalSDKTemplateInfo.cs [hunk 2]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.TemplateLocator
+{
+    public interface IOptionalSDKTemplateInfo
+    {
+        string TemplatePackageId { get; } // do we write down all templates package id and version?
```
Comment (wli3) [inline src/Microsoft.DotNet.TemplateLocator/IOptionalSDKTemplateInfo.cs:8]: yes packageid and version will be in the manifest.

File: src/Microsoft.DotNet.TemplateLocator/IOptionalSDKTemplateInfo.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.TemplateLocator
+{
+    public interface IOptionalSDKTemplateInfo
```
Comment (wli3) [inline src/Microsoft.DotNet.TemplateLocator/IOptionalSDKTemplateInfo.cs:6]: TemplatePackageInfo

File: src/Microsoft.DotNet.TemplateLocator/ITemplateLocator.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+
+namespace Microsoft.DotNet.TemplateLocator
+{
+    public interface ITemplateLocator
+    {
+        public IReadOnlyCollection<IOptionalSDKTemplateInfo> GetDotnetSdkTemplatesFromSDKversion(string SDKversion);
```
Comment (wli3) [inline src/Microsoft.DotNet.TemplateLocator/ITemplateLocator.cs:10]: GetDotnetSdkTemplatePackages(string sdkVersion)

File: src/Microsoft.DotNet.TemplateLocator/IOptionalSDKTemplateInfo.cs [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+    {
+        string TemplatePackageId { get; } // do we write down all templates package id and version?
+        string TemplateVersion { get; }
+        string Path { get; }
```
Comment (wli3) [inline src/Microsoft.DotNet.TemplateLocator/IOptionalSDKTemplateInfo.cs:10]: Path will come the library

File: src/Microsoft.DotNet.TemplateLocator/IOptionalSdkTemplatePackageInfo.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DotNet.TemplateLocator
+{
+    public interface IOptionalSdkTemplatePackageInfo
+    {
+        string TemplatePackageId { get; }
+        string TemplateVersion { get; }
```
Comment (wli3) [inline src/Microsoft.DotNet.TemplateLocator/IOptionalSdkTemplatePackageInfo.cs:9]: package id and package version is no really needed

File: src/Microsoft.DotNet.TemplateLocator/TemplateLocator.cs [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Reflection;
+
+namespace Microsoft.DotNet.TemplateLocator
+{
+    public sealed class TemplateLocator: ITemplateLocator
+    {
+        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(string sdKversion)
```
Comment (grinrag) [inline src/Microsoft.DotNet.TemplateLocator/TemplateLocator.cs:15]: sdkVersion

**Pull request closed without merging**